src/subnet.c

```diff
@@ -86,6 +147,32 @@ void free_subnet(subnet_t *subnet) {
 	free(subnet);
 }
 
+void subnet_cache_flush(subnet_t *subnet) {
+	switch(subnet->type) {
+	case SUBNET_IPV4:
+		if(subnet->net.ipv4.prefixlength == 32) {
+			hash_delete(ipv4_t, &ipv4_cache, &subnet->net.ipv4);
+			return;
+		}
+
+		break;
+
+	case SUBNET_IPV6:
+		if(subnet->net.ipv4.prefixlength == 128) {
+			hash_delete(ipv6_t, &ipv6_cache, &subnet->net.ipv6);
+			return;
+		}
+
+		break;
+
+	case SUBNET_MAC:
+		hash_delete(mac_t, &mac_cache, &subnet->net.mac);
+		return;
+	}
```

Comment:
My compiler now complains that we don't handle `SUBNET_TYPES` in this switch-statement. I think we should remove the `SUBNET_TYPES` enum value, and have an `subnet_cache_flush_tables(void)` function for if we want to flush all tables, drop `subnet_cache_flush_table(subnet_type_t)` since the only remaining user would be this function, and here we can just call `hash_clear()` directly.

Reply:
Updated